edgeless_telemetry/src/elasticsearch_api.rs

```diff
@@ -0,0 +1,203 @@
+use serde_json::Value;
+use url::Url;
+
+//static mut variable to hold the counter
+static mut COUNTER: u32 = 0;
+
+/// Establish connection to specified ES endpoint.
+/// # Returns
+/// A Result indicating success or failure.
+pub fn es_create_client() -> Result<Elasticsearch, Box<dyn std::error::Error>> {
+    //define ES endpoint configs
+
+    let url = Url::parse("elastic_endpoint_url")?; //contant Panagiotis Antoniou(Aegis) for details
+    let credentials = Credentials::Basic("username".into(), "password".into()); //contact Panagiotis Antoniou(Aegis) for details
+    let conn_pool = SingleNodeConnectionPool::new(url);
+    let transport = TransportBuilder::new(conn_pool).auth(credentials).build()?;
+
+    Ok(Elasticsearch::new(transport))
+}
+
+/// Creates an Elasticsearch index with a specified mapping.
+/// # Arguments
+/// * `client_result` - the Elasticsearch client or an error.
+/// # Returns
+/// A Result indicating success or failure.
+//To be called once on init.
+pub async fn es_create_index(
+    client_result: &Result<Elasticsearch, Box<dyn std::error::Error>>,
+    index_flag: bool,
+) -> Result<(), Box<dyn std::error::Error>> {
+    //define mapping
+    let client = match client_result {
+        Ok(client) => client,
+        Err(err) => return Err(err.to_string().into()), // Convert error to String
+    };
+    //define ES index mapping based on the index_flag value
+    let mapping = if index_flag {
+        //create edgeless_resources mapping
+        json!({
+            "mappings": {
+                "properties": {
+                    "cpu_percent": { "type": "float" },
+                    "memory_percent": { "type": "float" },
+                    "timestamp": { "type": "date" }
+                }
+            }
+        })
+    } else {
+        //create edgeless_runtime mapping (currently based on stdout console log)
+        json!({
+            "mappings": {
+                "properties": {
+                    "event": { "type": "keyword" },
+                    "timestamp": {"type": "date"},
+                    "tags": {
+                        "properties": {
+                            "FUNCTION_ID": { "type": "keyword" },
+                            "FUNCTION_TYPE": { "type": "keyword" },
+                            "NODE_ID": { "type": "keyword" }
+                        }
+                    }
+                }
+            }
+        })
+    };
+
+    let create_index_response = client
+        .indices()
+        .create(IndicesCreateParts::Index(get_index_name(index_flag)))
+        .body(mapping)
+        .send()
+        .await?;
+
+    if create_index_response.status_code().is_success() {
+        println!("Index 'edgeless_runtime' created successfully");
+    } else {
+        println!("Failed to create index 'edgeless_runtime': {}", create_index_response.status_code());
+    }
+    Ok(())
+}
+
+//static counter for indexing identification
+fn increment_counter() -> String {
+    unsafe {
```

Comment:
No unsafe code is allowed. Guess you need that for mutability, which you need to ensure using a mutex.
Also probably you don't want a static here.